pkg/virt-handler/vm.go

```diff
@@ -21,6 +22,18 @@ func NewVMController(listWatcher cache.ListerWatcher, domainManager libvirt.Doma
 				recorder.Event(vm, kubev1.EventTypeWarning, v1.SyncFailed.String(), err.Error())
 				return cache.ErrRequeue{Err: err}
 			}
+
+			restClient, err := kubecli.GetRESTClient()
+			if err != nil {
+				return cache.ErrRequeue{Err: err}
+			}
+			vm.Status.Phase = v1.Succeeded
```

Comment:
When modifying an object, please copy it first. Otherwise you are directly manipulating  the object in the cache.

Also have a look at https://github.com/kubernetes/community/blob/master/contributors/devel/controllers.md,

we are not yet there because `workqueue` was only added to upstream kubernetes sdk, but whatever we do, we want to fit into this way of writing our processing loops.